src/components/SingleArticle.jsx

```diff
@@ -1,16 +1,69 @@
 import { useParams } from "react-router-dom";
 import { useLocation } from "react-router-dom";
+import { useState, useEffect } from "react";
 import CommentCard from "./CommentCard";
+import { patchArticles, getArticles } from "../api";
 
 function SingleArticle() {
   const { article_id } = useParams();
-  const location = useLocation();
-  const articles = location.state;
+  const [article, setArticle] = useState({})
+  const [articleVote, setArticleVote] = useState(0)
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState();
 
 
-  for (let i = 0; i < articles.length; i++) {
-    if (article_id == articles[i].article_id) {
-      const article = articles[i];
+useEffect(()=>{
+  getArticles(article_id).then((response)=>{
+    setArticle(response.article)
+    setLoading(false);
+  })
+  .catch((error)=>{
+    setError("Error when getting articles");
+        setLoading(false);
+  })
+
+},[article_id])
+
+useEffect(()=>{
+    if (articleVote !== 0) {
```

Comment:
Same as above here, zero is a falsy value, so you are just checking if articleVote is truthy